core/src/main/java/hudson/model/Node.java

```diff
@@ -451,6 +451,46 @@ public FileSystemProvisioner getFileSystemProvisioner() {
      */
     public abstract @Nonnull DescribableList<NodeProperty<?>, NodePropertyDescriptor> getNodeProperties();
 
+    /**
+     * Gets the specified property or null if the property is not configured for
+     * this Node.
+     * 
+     * @param clazz the type of the property
+     * 
+     * @return null if the property is not configured
+     * 
+     * @since TODO
+     */
+    @CheckForNull
+    public <T extends NodeProperty> T getNodeProperty(Class<T> clazz)
+    {
+        for (NodeProperty p: getNodeProperties()) {
```

Comment:
This cycle has very confusing brackets. Could you please add brackets to `if` and properly tabulate the last bracket?